### PR TITLE
ES|QL Add serialization to fulltext predicate types

### PR DESCRIPTION
**Pull request by ChrisHegarty**
This commit adds serialization to fulltext predicate types, so that queries build atop these types can be sent to data nodes.

### CODE REVIEW

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine update branch

**Comment by ChrisHegarty** [comment]
@elasticmachine run elasticsearch-ci/8.15.0/ bwc-snapshots

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/fulltext/MultiMatchQuerySerializationTests.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            randomOptionOrNull()
+        );
+    }
+
+    @Override
+    protected MultiMatchQueryPredicate mutateInstance(MultiMatchQueryPredicate instance) throws IOException {
+        var fieldString = instance.fieldString();
+        var query = instance.query();
+        var options = instance.options();
+        switch (randomFrom(List.of(1, 2, 3))) {
```
Comment (nik9000) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/fulltext/MultiMatchQuerySerializationTests.java:37]: I usually use `between(0, 2)` for this. I wonder if this is more clear.... I dunno!

**Review by nik9000 — APPROVED** [review]

**Review by tteofili — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**